prereise/gather/hiflddata/data_trans.py

```diff
@@ -0,0 +1,600 @@
+
+
+def write_sub(clean_data, zone_dic):
+    """Write the data to sub.csv as output
+
+    :param pandas.DataFrame clean_data: substation dataframe as returned by :func:`clean`
+    :param dict zone_dic: zone dict as returned by :func:`get_zone`
+    """
+
+    output = pd.DataFrame()
+    output["sub_id"] = clean_data["ID"]
+    output["sub_name"] = clean_data["NAME"]
+    output["lat"] = clean_data["LATITUDE"]
+    output["lon"] = clean_data["LONGITUDE"]
+    output["zone_id"] = clean_data.apply(lambda x: zone_dic[x["STATE"]], axis=1)
+    output["type"] = clean_data["TYPE"]
+    output.to_csv("output/sub.csv", index=False)
+
+
+def write_bus(clean_data, zone_dic, kv_dict):
+    """Write the data to bus.csv as output
+
+    :param pandas.DataFrame clean_data: substation dataframe as returned by :func:`clean`
+    :param dict zone_dic: zone dict as returned by :func:`get_zone`
+    :param dict kv_dict: substation KV dict
+    """
+
+    missing_sub, data = [], []
+    for _, row in clean_data.iterrows():
+        sub = (row["LATITUDE"], row["LONGITUDE"])
+        if sub in kv_dict:
+            data.append((row["ID"], 0, zone_dic[row["STATE"]], kv_dict[sub]))
+        else:
+            missing_sub.append(sub)
+
+    output = pd.DataFrame(data, columns=["bus_id", "pd", "zone_id", "base_kv"])
+    output.to_csv("output/bus.csv", index=False)
+
+    print(
+        "INFO: ",
+        len(missing_sub),
+        " substations excluded from the network. Some examples:",
+    )
+    print(missing_sub[:20])
+
+
+def write_bus2sub(clean_data):
+    """Write the data to bus2sub.csv as output
+
+    :param pandas.DataFrame clean_data: substation dataframe as returned by :func:`clean`
+    """
+
+    output = pd.DataFrame()
+    output["bus_id"] = clean_data["ID"]
+    output["sub_id"] = clean_data["ID"]
+    output.to_csv("output/bus2sub.csv", index=False)
+
+
+def write_branch(lines):
+    """Write the data to branch.csv as output
+
+    :param list lines:  a list of lines as returned by :func:`neighbors`
+    """
+
+    lines.to_csv("output/branch.csv", index=False)
+
+
+# Reactance Calculate 1
+kv_xperunit_calculate_1 = (
+    (69, 0.2293),
+    (115, 0.1106),
+    (230, 0.0567),
+    (345, 0.0074),
+    (500, 0.0001),
+    (1100, 0.0001),
+)
+
+# Reactance Calculate 2
+kv_xperunit_calculate_2 = (
+    (69, 0.0096),
+    (100, 0.0063),
+    (115, 0.0039),
+    (138, 0.0026),
+    (161, 0.0021),
+    (230, 0.0011),
+    (345, 0.0005),
+    (500, 0.0002),
+    (765, 0.0001),
+    (1100, 0.0001),
+)
+
+# RateA Calculate 3
+kv_rate_a_calucate_3 = (
+    (69, 86),
+    (100, 181),
+    (115, 239),
+    (138, 382),
+    (161, 446),
+    (230, 797),
+    (345, 1793),
+    (500, 2598),
+    (765, 5300),
+    (1100, 5300),
+)
+
+# RateA Calculate 3
+kv_sil_calucate_3 = (
+    (69, 13),
+    (100, 30),
+    (115, 35),
+    (138, 50),
+    (161, 69),
+    (230, 140),
+    (345, 375),
+    (500, 1000),
+    (765, 2250),
+    (1100, 2250),
+)
```

Comment:
This variable seems to never be used, but we use `for kv, sil in kv_rate_a_calucate_3` within `compute_rate_a`. Should we be using this variable instead of `kv_rate_a_calucate_3`?

Reply:
That is a good catch! This should be applied when the line length>=50 miles. Will correct the copy-paste error.